components/ai_chat/content/browser/ai_chat_brave_search_throttle.cc

```diff
@@ -0,0 +1,159 @@
+/* Copyright (c) 2024 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+#include "brave/components/ai_chat/content/browser/ai_chat_brave_search_throttle.h"
+
+#include <memory>
+#include <utility>
+
+#include "base/check.h"
+#include "base/functional/bind.h"
+#include "brave/components/ai_chat/content/browser/ai_chat_tab_helper.h"
+#include "brave/components/ai_chat/content/browser/page_content_fetcher.h"
+#include "brave/components/ai_chat/core/browser/conversation_handler.h"
+#include "brave/components/ai_chat/core/browser/utils.h"
+#include "brave/components/ai_chat/core/common/utils.h"
+#include "components/user_prefs/user_prefs.h"
+#include "content/public/browser/browser_context.h"
+#include "content/public/browser/navigation_handle.h"
+#include "content/public/browser/permission_controller.h"
+#include "content/public/browser/permission_request_description.h"
+#include "content/public/browser/web_contents.h"
+
+namespace ai_chat {
+
+// static
+std::unique_ptr<AIChatBraveSearchThrottle>
+AIChatBraveSearchThrottle::MaybeCreateThrottleFor(
```

Comment:
what happens if the user is in a private tab btw? presumably the search page would still show the "continue" button, but clicking it just does nothing?

Reply:
The throttle won't be created for non-regular profile, checks in browser/brave_content_browser_client.cc.
So yes, they still show the button, and it will just trying to navigate to that page (as what it was without the throttle), hence a 404 page.
<img width="1096" alt="Screenshot 2024-10-23 at 11 31 27 AM" src="https://github.com/user-attachments/assets/1f8a8481-c7bc-470d-a1bf-6139c37e631e">